packages/back-end/src/events/handlers/slack/slack-event-handler-utils.ts

```diff
@@ -284,6 +291,75 @@ const buildSlackMessageForExperimentDeletedEvent = (
   };
 };
 
+const buildSlackMessageForExperimentWarningEvent = ({
+  data,
+}: ExperimentWarningNotificationEvent): SlackMessage => {
+  let invalidData: never;
+
+  switch (data.type) {
+    case "auto-update-failed": {
+      const text = `Automatic snapshot creation for ${data.experimentName} failed!`;
+
+      return {
+        text,
+        blocks: [
+          {
+            type: "section",
+            text: {
+              type: "mrkdwn",
+              text:
+                `Automatic snapshot creation for *${data.experimentName}* failed!` +
+                getExperimentUrlFormatted(data.experimentId),
+            },
+          },
+        ],
+      };
+    }
+
+    case "multiple-exposures": {
+      const baseText = `${data.usersCount} users (${data.percent}%) saw multiple variations and were automatically removed from results. Check for bugs in your implementation, event tracking, or data pipeline.`;
+
+      return {
+        text: `Multiple Exposures Warning for experiment ${data.experimentName}: ${baseText}`,
+        blocks: [
+          {
+            type: "section",
+            text: {
+              type: "mrkdwn",
+              text:
+                `Multiple Exposures Warning for experiment *${data.experimentName}*: : ${baseText}` +
+                getExperimentUrlFormatted(data.experimentId),
+            },
+          },
+        ],
+      };
+    }
+
+    case "srm": {
+      const baseText = `P-value below ${data.threshold}.`;
```

Comment:
Minor copy suggestion:

`Traffic imbalance detected for experiment ${data.experimentName}: Sample Ratio Mismatch (SRM) p-value below ${data.threshold}.`